com.ibm.streamsx.topology/opt/python/packages/streamsx/rest_primitives.py

```diff
@@ -1160,10 +1250,119 @@ def get_view_obj(_view, rc):
                     return view
     return None
 
-
 class StreamingAnalyticsService(object):
     """Streaming Analytics service running on IBM Bluemix cloud platform.
     """
+    def __init__(self, rest_client, credentials):
+        # If IAM, create a V2 delegator, if basic http auth, create a V1 delegator
+        #
+        # Delegators are required because we need to keep StreamingAnalyticsService
+        # around for backwards compatibility, yet it also needs to work with basic
+        # and IAM authentication.
+        if 'v2_rest_url' in credentials:
```

Comment:
I'm not sure this is correct. `_StreamingAnalyticsServiceV2Delegator` always seems to assume a IAM rest client but I think it can be passed a user/password REST client.

I think `_StreamingAnalyticsServiceV2Delegator` always assumes IAM because it always calls `_get_authorization` which doesn't seem to exist in the non-IAM client.

Reply:
Right, so instead, it should look like:
```python
if 'username' in credentials and 'password' in credentials:
    # use v1
else:
    # use v2
```

Reply:
Would you agree?

Reply:
I added a `_get_authorization` method to _StreamsRestClient:
```python
    def _get_authorization(self):
        return requests.auth._basic_auth_str(self.username, self.password)
```

Reply:
Isn't that a fixed value for v1, that could be calculated once in `__init__`?

Reply:
Fixed.